### PR TITLE
Added post_upgrade marker to tier_marks list

### PR DESCRIPTION
**Pull request by OdedViner**
#7540
For example here, we dont need to run the tests because ceph status is not `health_ok`

https://url.corp.redhat.com/9cdaeac

### CODE REVIEW

**Comment by petr-balogh** [comment]
Hello @OdedViner , didn't get the need for this PR?

To have tier defined which will run after upgrade and what we have in our production jobs is something called: 
 tier_after_upgrade.

This is in scenario we run:
Deployment
Upgrade (pre_upgrade, upgrade, post_upgrade tests)
Tier execution after the successful upgrade run above (We run with: `tier1 or tier_after_upgrade`)

Post upgrade marker is already defined here:
https://github.com/red-hat-storage/ocs-ci/blob/460df6c632e0753aaa18191b7cfa7c14c8903f93/ocs_ci/framework/pytest_customization/marks.py#L116

The value of your post_upgrade will anyway got overwritten by the definition on line 116, no?

**Comment by OdedViner** [comment]
> Hello @OdedViner , didn't get the need for this PR?
> 
> To have tier defined which will run after upgrade and what we have in our production jobs is something called: tier_after_upgrade.
> 
> This is in scenario we run: Deployment Upgrade (pre_upgrade, upgrade, post_upgrade tests) Tier execution after the successful upgrade run above (We run with: `tier1 or tier_after_upgrade`)
> 
> Post upgrade marker is already defined here:
> 
> https://github.com/red-hat-storage/ocs-ci/blob/460df6c632e0753aaa18191b7cfa7c14c8903f93/ocs_ci/framework/pytest_customization/marks.py#L116
> 
> The value of your post_upgrade will anyway got overwritten by the definition on line 116, no?

we dont check health because post_upgrade not in tier_mark_name list https://github.com/red-hat-storage/ocs-ci/blob/master/tests/conftest.py#L1392
https://github.com/red-hat-storage/ocs-ci/blob/master/tests/conftest.py#L1462

**Comment by petr-balogh** [comment]
Then the fix should rather look like that you will define another set of the upgrade_markers for all of those:
https://github.com/red-hat-storage/ocs-ci/blob/4ba411d5b41c4f159d17ed62279d3fd0323053a4/ocs_ci/framework/pytest_customization/marks.py#L109-L116
```
upgrade_marks = [
    ocp_upgrade,
    ocs_upgrade,
    # the rest of those from link above
]```

And on the line 
https://github.com/red-hat-storage/ocs-ci/blob/master/tests/conftest.py#L1401
I would do it like:
```
    for each_tier in tier_marks + upgrade_marks:
```
to combine those tier_marks with upgrade marks.

Than maybe we can rename whole tier_marks_name fixture to:
tier_and_upgrade_marks_name.

As I think we would like to have this health check not just for post upgrade tests but for all upgrade related tests as well.

**Comment by OdedViner** [comment]
Running oded_test:
https://url.corp.redhat.com/8b8e8d2

File: tests/conftest.py [hunk 1]
```diff
@@ -1406,8 +1407,26 @@ def tier_marks_name():
...
+    Returns:
+        list: list of upgrade mark names
+
+    """
+    upgrade_marks_name = []
+    for upgrade_mark in upgrade_marks:
+        try:
+            upgrade_marks_name.append(upgrade_mark().args[0].name)
+        except AttributeError:
+            log.error(f"upgrade mark does not exist")
```
Comment (petr-balogh) [inline tests/conftest.py:1424]: tests/conftest.py:1424:23: F541 f-string is missing placeholders

Remove f please if you are not expanding any variable here.

**Comment by petr-balogh** [comment]
> @OdedViner @petr-balogh Could we prioritize this PR? The issue has been consistent in many runs.

Pinged Oded to remove the tests and fix flake8 issue.

**Comment by OdedViner** [comment]
> > @OdedViner @petr-balogh Could we prioritize this PR? The issue has been consistent in many runs.
> 
> Pinged Oded to remove the tests and fix flake8 issue.

Done. Running job https://url.corp.redhat.com/dbd2eae

**Review by am-agrawa — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**